litespi/core/xip.py

```diff
@@ -0,0 +1,103 @@
+from migen import *
+from migen.genlib.fsm import FSM, NextState
+
+from litex.soc.interconnect import wishbone, stream
+from litex.gen.common import reverse_bytes
+
+from litespi.common import *
+
+
+class LiteSPIXIP(Module):
```

Comment:
I wonder if `LiteSPIMemoryMapped` might be a better name?

Reply:
I admit `xip` is a bit confusing, instead of `xip` in the code we could use `mmap` and for this module LiteSPIMMAP?